Blog post about GAP lesson

olexandr-konovalov opened the pull request:
This is a blog post on the GAP Software Carpentry lesson (listed as "contributed" at http://software-carpentry.org/lessons/): 

- lesson itself: http://alex-konovalov.github.io/gap-lesson/
- published via Zenodo: http://doi.org/10.5281/zenodo.167362
- lesson repository: https://github.com/alex-konovalov/gap-lesson

It happened that in the process of writing the blog post became bigger than I initially intended - please let me know if you'd like any revisions.

Commit:
First draft of the blog post on the GAP lesson

Commit:
More on the structure of the lesson + proofreading

Commit:
More on the week-long course

Commit:
Intro paragraph - what is this post about

alex-konovalov committed:
Removed accidentally repeated text

alex-konovalov committed:
Another typo

gvwilson commented on _posts/2016/11/2016-11-21-gap-lesson.md:
```diff
@@ -0,0 +1,151 @@
+---
+
+This is a blog post about the Software Carpentry lesson
+["Programming with GAP"](http://alex-konovalov.github.io/gap-lesson/).
```
This sentence feels a bit lonely on its own - maybe something like, "Software Carpentry is more than just a set of workshops and lessons. It is also a way to *develop* lessons, one that we have used successfully to create a lesson on [Programming with GAP](...)."

olexandr-konovalov replied:
Thanks @gvwilson - done

gvwilson commented on _posts/2016/11/2016-11-21-gap-lesson.md:
```diff
@@ -0,0 +1,151 @@
+[here](http://www.codima.ac.uk/2015/07/01/average-order-of-group-elements-a-demo-of-test-driven-development-in-gap/)),
+an thus the choice has been made.
+
+Indeed, the problem of determining an average order of the element of the group
```
I think the following four paragraphs are more detail than our readers will want - can they be compressed to a single paragraph with pointers to the original (longer) blog post?

Commit:
Intro paragraph as suggested by @gvwilson

Commit:
Shortened lesson overview as suggested by @gvwilson

The more detailed lesson overview is now availble on my homepage at
http://blogs.cs.st-andrews.ac.uk/alexk/2016/11/22/publishing-software-carpentry-lesson-on-gap/
